### ISSUE TITLE
SmartThings built in Sync with Repo

### ISSUE THREAD

**Issue by snakedog116** [yracine/device-type.myecobee#16]
Hello,

First of all, I've been using your Ecobee device and smartapps for a couple of weeks and am so impressed with your work. Thank you very much for developing and sharing this. And also thanks for being very active to keep it up to date.

I'm new to GitHub and SmartThings, so perhaps I'm missing something, but I'm trying to find a way to keep things updated. (Sorry if this isn't the right place to ask about this. I couldn't find a way to PM on GitHub)

What I'm wondering is if it's possible to re-organize the files for this repository so that it fits SmartThings' file structure. The reason for that is because SmartThings has a way to quickly download from a GitHub repository within the graph.api.smartthings.com site. However I'm not able to sync your repository with my SmartThings account because apparently the file structure doesn't match what SmartThings is wanting to see. Seems odd, but that's what I was told on the SmartThings Community site. https://community.smartthings.com/t/github-integration-update-from-repo-error/31118/8

I see you have a SmartThingsPublic repository, and I can sync with that, but the ecobee devices and smartaps aren't on there.

Sorry if this is a dumb question and I should be able to do this on my end somehow, but like I said, I'm new to all of this.

Thanks, I appreciate any help.

And I also understand that re-organizing all the files may not be worth your time and I appreciate and understand if that's your reply.

**Comment by StrykerSKS** [comment]
I brought this up on the community site. I was told there is an interest if the steps to make the change were outlined. I was going to take a stab at it last weekend but didn't get to it. But I'm off all next week for Christmas break so was going to give it a shot.

**Comment by snakedog116** [comment]
Oh awesome! Thank you so much!

**Comment by StrykerSKS** [comment]
Oops, I'm actually off the week after, but will still try to get to it by next weekend if I can.

**Comment by yracine** [comment]
Hi,  my code is not "open source", so please respect the license..

**Comment by snakedog116** [comment]
@yracine, I'm sorry did I ask for something wrong? Or are you talking to @StrykerSKS?

Is there an issue with someone taking your codes and reorganizing it into different folders? Would that be against your license?

Are you able to reorganize your folder so that @StrykerSKS doesn't have to?

**Comment by StrykerSKS** [comment]
Yes, of course! I've temporarily forked your base to determine the changes
needed. Once I create the patches I'll create a Pull request and pass back
to you. I don't think I will need to change any actual code, just file
names, paths, etc. I'll hand it all back to you and then delete my forked
repository.

While I prefer open source, I fully respect your choice to use whatever
license you see fit and will certainly honor that.

// Sean

**Comment by yracine** [comment]
Hi @StrykerSKS, @snakedog116 Just to let you know, my decision is now not to integrate with the SmartThingsPublic directory.... The reason is that SmartThings is a sort of unstable at the moment (especially with my code) and I do no want to called for more support.  I have a full time job and a family, and I do not want to spend more time supporting ST users.  So, please respect my choice and leave my repository isolated from ST.

**Comment by StrykerSKS** [comment]
As you wish, but making the changes I suggest does not make it
automatically visible in the Public SmartThings directory. It only makes it
possible for people that use your module to *manually* add it to their
instance of the IDE so that when you make changes to your repository they
simply sync those changes. This would, in fact, REDUCE the amount of
support since users would automatically see that they don't have the latest
version and prompt them to update. Instead of you having to always say
"please update to the latest version"...

Very disappointing choice in my mind. You should really consider your
choice of license as a possible contributor to this support pain. If you
had a more open license then you'd possibly garner more community
supporters that could take up some of the load.

I guess I should start my search for a new Ecobee device type and smart
apps. No point being on a codebase that will not be supported.

Best Regards,
Sean

**Comment by yracine** [comment]
I'll let you know if I change my mind, but for the moment, I prefer not to be integrated.

**Comment by frankhinek** [comment]
I respect your right to choose how to distribute your code, but after your clarification that your code is not "open source" I feel obligated to discontinue use and delete from my ST hub.

**Comment by snakedog116** [comment]
@yracine This topic ended up a bit off topic, are you able to re-organize this repo so that users can more easily update their apps?

This doesn't mean you have to upload and make it open source on the SmartThingsPublic repo, you could just temporarily upload it to your personal SmartThingsPublic repo. This would create the folder and file structure required for repo sync within the IDE. Then you can copy those files/folders back into this repo, delete it from your personal SmartThinsPublic repo and it wouldn't be open source, none of your licensing would change, but it would let users like myself who really appreciate the work you've done on this to much more quickly see that my code is out of date and update it. I do think it would help with how much time you are helping troubleshoot.


For example, take a look at this repo from @justintime. he re-organized his repo to make it sync within the IDE and it's so helpful.

Old Repo (Doesn't sync): https://github.com/justintime/SmartThings-Monoprice-11990
New Repo (Dows sync): https://github.com/justintime/ST-Monoprice-Dual-Relay-Module

**Comment by yracine** [comment]
Hi, I may do it later when I will have some time... At the moment, this is not a priority for me.

Also, please note that any new versions of my Ecobee device handler will no longer be available on github.

Only the ones who contribute financially to Maisons Ecomatiq will receive the new code.

http://thingsthataresmart.wiki/index.php?title=My_Ecobee_Device#Notes_to_MyEcobee_Device_Users_-_Contribution_is_now_required

Regards.

**Comment by StrykerSKS** [comment]
This is quite an unfortunate turn of events. I've contributed to many
developers and projects over the years but given this new turn of events
I've decided that this is not a good situation for me as an end-user (hence
why I prefer open source).

So, with that in mind, I've gone back to the original devicetype and
smartapp available in the public repository. I've spent the last few days
working on getting it to work better. Once I've got a semi stable version
available I'll post something in the general forums.

Until then, if anyone wants to jump on board let me know and I can ping you
when I have something worth sharing.

// StrykerSKS
